ontopy/ontology.py

```diff
@@ -1490,22 +1499,23 @@ def addancestors(entity, counter, subject):
                     subject.add(parent)
                     addancestors(parent, counter - 1, subject)
 
-        if isinstance(include, str) and include.isdigit():
-            include = int(include)
-
-        if include == "all":
-            ancestors.update(*(_.ancestors() for _ in classes))
-        elif include == "closest":
-            closest = self.closest_common_ancestor(*classes)
+        if sum(map(bool, [closest, generations])) > 1:
+            raise ValueError(
+                "Only one of `generations` or `closest` may be specified."
+            )
+        if closest:
+            closest_ancestor = self.closest_common_ancestor(*classes)
             for cls in classes:
                 ancestors.update(
-                    _ for _ in cls.ancestors() if closest in _.ancestors()
+                    _
```

Comment:
The anonymous variable name `_` should be reserved for cases where we explicitly are ignore a variable. For instance, if we have a list of triples and only are interested in looping over the subjects, it is ok to write:

```python
triples = [('s1', 'p1', 'o1'), ('s2', 'p2', 'o2')]
for s, _, _ in triples:
    # so something with `s`
```

Please note that `_` also has a special meaning in the Python interpreter - it is always assigned to the latest result. For example

```python
>>> 1 + 1
2
>>> _
2
```
Starting to use it as a variable can create confusion - for instance if someone wants to debug your code and paste your code lines into the interpreter, statement for statement, the value of `_` will be overwritten by the interpreter...

In this case I would use name `a` as a good, short and readable name for the temporary variable. That it is a single letter carries the semantics of it being a temporary variable.

Reply:
Tahnk you for really informative teaching!